src/Sylius/Bundle/LocaleBundle/Form/DataTransformer/LocaleToCodeTransformer.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\LocaleBundle\Form\DataTransformer;
+
+use Sylius\Component\Locale\Model\LocaleInterface;
+use Sylius\Component\Locale\Provider\LocaleCollectionProviderInterface;
+use Symfony\Component\Form\DataTransformerInterface;
+
+/** @phpstan-ignore-next-line */
+final class LocaleToCodeTransformer implements DataTransformerInterface
+{
+    public function __construct(private LocaleCollectionProviderInterface $localesProvider)
+    {
+    }
+
+    public function transform(mixed $value): string
+    {
+        if (!$value instanceof LocaleInterface) {
+            throw new \InvalidArgumentException(
```

Comment:
Would be cool to use specialized exception for DataTransformer.

https://github.com/Sylius/Sylius/blob/9e0d60a3152b6ac6d1686e5073bc3baa35f23ca0/src/Sylius/Bundle/ProductBundle/Form/DataTransformer/ProductVariantToProductOptionsTransformer.php#L38-L40

Reply:
ECS will fix that.

Reply:
Lol, when reading it for the first time, I thought it was only about formatting :|. I'll check that.